crates/red_knot_python_semantic/src/types.rs

```diff
@@ -273,7 +273,8 @@ pub enum Type<'db> {
     /// A heterogeneous tuple type, with elements of the given types in source order.
     // TODO: Support variable length homogeneous tuple type like `tuple[int, ...]`.
     Tuple(TupleType<'db>),
-    // TODO protocols, callable types, overloads, generics, type vars
+    TypeVar(TypeVarInstance<'db>),
```

Comment:
Maybe add some documentation here, or is the documentation on `TypeVarInstance`?

Reply:
Done.  (There is documentation down at `TypeVarInstance` describing the fields of the struct, but I added some here too describing the intent of this type variant)